src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgrade.cs

```diff
@@ -22,6 +26,9 @@ public class ConsoleUpgrade : IAppCommand
         private readonly ITelemetry _telemetry;
         private readonly IUpgradeStateManager _stateManager;
         private readonly ILogger<ConsoleUpgrade> _logger;
+        private readonly IAnalyzeResultWriterProvider _writerProvider;
+        private readonly IUpgradeResultCollector _upgradeResultCollector;
+        private readonly IOptions<AnalysisOptions> _analysisOptions;
```

Comment:
Looks like `AnalysisOptions` should be separated into what is actually options for analysis, and also `OutputOptions` or something for options around output

Reply:
Agreed, I think when I refactored for Binary Analysis I put the `.Format` property on a base class, the Upgrade options could/should now inherit from that one so it gets the `--format` arg available to it and exposed via UpgradeOptions